python/ray/experimental/workflow/virtual_actor.py

```diff
@@ -0,0 +1,350 @@
+        DerivedActorClass.__qualname__ = name
+        # Construct the base object.
+        self = DerivedActorClass.__new__(DerivedActorClass)
+
+        self._metadata = metadata
+        return self
+
+    def create(self, *args, **kwargs):
+        """Create an actor.
+
+        Args:
+            args: These arguments are forwarded directly to the actor
+                constructor.
+            kwargs: These arguments are forwarded directly to the actor
+                constructor.
+
+        Returns:
+            A handle to the newly created actor.
+        """
+        return self._create(
+            args=args,
+            kwargs=kwargs,
+            actor_id=None,
+            storage=None,
+            readonly=False)
+
+    def get(self, actor_id: str, storage: Storage):
+        """Get an existing virtual actor yby inheriting its state."""
+        return self._get(actor_id=actor_id, storage=storage, readonly=False)
+
+    def options(self,
+                actor_id: str,
+                storage: Optional[Storage] = None,
+                readonly: bool = False):
+        """Configures and overrides the actor instantiation parameters.
+
+        Examples:
+
+        .. code-block:: python
+
+            @workflow.actor
+            class Foo:
+                def method(self):
+                    return 1
+            # By default class Foo will not be running in the readonly mode,
+            # and will use a randomly generated 'actor_id'.
+            # You can specify 'readonly' and 'actor_id' by
+            Bar = Foo.options(actor_id="my_actor_id", readonly=True)
+        """
+
+        actor_cls = self
+
+        class ActorOptionWrapper:
+            def create(self, *args, **kwargs):
+                return actor_cls._create(
+                    args=args,
+                    kwargs=kwargs,
+                    actor_id=actor_id,
+                    storage=storage,
+                    readonly=readonly)
+
+            def get(self):
+                return actor_cls._get(
+                    actor_id=actor_id, storage=storage, readonly=readonly)
+
+        return ActorOptionWrapper()
+
+    def _create(self, args, kwargs, actor_id: Optional[str],
+                storage: Optional[Storage], readonly: bool):
+        function_signature = self._metadata.signatures["__init__"]
+        # flatten and check args
+        creation_args = signature.flatten_args(function_signature, args,
+                                               kwargs)
+        if actor_id is None:
+            actor_id = self._metadata.generate_random_actor_id()
+        if storage is None:
+            storage = get_global_storage()
+        instance = VirtualActor(self._metadata, actor_id, storage, readonly)
+        instance._create(creation_args)
+
+    def _get(self, actor_id: str, storage: Storage, readonly: bool):
+        return VirtualActor(self._metadata, actor_id, storage, readonly)
+
+
+class VirtualActor:
```

Comment:
To be consistent with ray, let's rename it to `VirtualActorHandler`

Reply:
Since we cannot pass virtual actor around (e.g. as a function argument), it should not be a handle to me (similarly, a workflow should not just be a task/objectref)